src/main/java/training/endava/skeleton/Main.java

```diff
@@ -1,118 +1,148 @@
 
-             nrOfTags.get(size).add(a);
-        }
 
-        for (Integer key : nrOfTags.keySet()) {
-            System.out.println("Key = " + key);
-            System.out.println("List =" + nrOfTags.get(key));
-        }
+        //System.out.println("LIST 1:");
+        //System.out.println(personList);
+
+        MockDB mockDB = MockDB.getInstance();
+        mockDB.setTable(Person.class, personList);
+
+        //System.out.println("LIST 2:");
+        //System.out.println(mockDB.getTable(Person.class));
+
+        PersonRepository personRepository = new PersonRepository();
+
+        personRepository.save(new Person("Ana09", 10, "0706", "Iasi06") );
+        personRepository.findById(10);
+
+        //System.out.println(personRepository.existsById(10));
+
+        Company fisrtCompany = new Company("C01", 1987,personList, 100,CompanyArea.IT);
+        Company secondCompany = new Company("C02", 2014,personList2,200,CompanyArea.QA);
+        Company thirdCompany = new Company("C03", 1875,personList3,300,CompanyArea.FINANCE);
+
+        List<Company> companyList = new ArrayList<>();
+        companyList.add(fisrtCompany);
+        companyList.add(secondCompany);
+        companyList.add(thirdCompany);
+
+        List<String> names = companyList.stream()
+                .map(Company::getName)
+                .collect(Collectors.toList());
+        //System.out.println(names);
+
+        int maxCapacity = companyList.stream()
+                .mapToInt(Company::getCompanyCapacity)
+                .max()
+                .getAsInt();
+
+        //System.out.println(maxCapacity);
+
+        Company oldestCompany = companyList.stream()
+                .max((c1, c2) -> Integer.compare(c1.getFoundationYear(),(c2.getFoundationYear())))
+                .get();
 
-        LinkedHashMap<Integer, List<Article>> reverseSortedMap = new LinkedHashMap<>();
+        //System.out.println(oldestCompany);
 
+        int count = companyList.stream()
+                .map(Company::getPersons)
+                .map(List::size)
+                .reduce(0,(pers1 , pers2) -> pers1 + pers2)
+                .intValue();
 
-        nrOfTags.entrySet()
-                .stream()
-                .sorted(Map.Entry.comparingByKey(Comparator.reverseOrder()))
-                .forEachOrdered(x -> reverseSortedMap.put(x.getKey(), x.getValue()));
+        //System.out.println(count);
 
-        //System.out.println("Reverse Sorted Map   : " + reverseSortedMap);
-        for (Integer key : reverseSortedMap.keySet()) {
-            System.out.println("Key = " + key);
-            System.out.println("List =" + nrOfTags.get(key));
+        //System.out.println(nameConcatenation);
+
+        Person nameConcatenation = companyList.stream()
+                .map(Company::getPersons)
+                .flatMap(List<Person>::stream)
+                .reduce(new Person(), (per1, per2) -> {per1.setName(per1.getName() + per2.getName()); return per1;});
```

Comment:
Have you verified the resulted name? per1.getName() for the identity value will be null, so the resulted person will have the name starting with "null". I saw you have a constructor that has only the name as parameter, use it and initialize the name with empty string.